src/mscorlib/shared/System/Buffers/MemoryManager.cs

```diff
@@ -38,6 +33,20 @@ public abstract class MemoryManager<T> : IMemoryOwner<T>, IPinnable
         /// </summary>
         public abstract void Unpin();
 
+        /// <summary>
+        /// Returns a <see cref="System.Memory{T}"/> for the current <see cref="MemoryManager{T}"/> .
+        /// </summary>
+        /// <param name="length">The element count in the memory, starting at offset 0</param>
+        protected Memory<T> CreateMemory(int length) => new Memory<T>(this, length);
+
+        /// <summary>
+        /// Returns a handle to the memory that has been pinned and hence its address can be taken.
+        /// </summary>
+        /// <param name="start">The offset to the element which the returned memory starts at</param>
+        /// <param name="length">The element count in the memory, starting at element offset <paramref name="start"/></param>
+        /// <remarks>Convenience api for non-zero offsets</remarks>
+        protected Memory<T> CreateMemory(int start, int length) => CreateMemory(length).Slice(start);
```

Comment:
Use a Memory ctor instead of slicing.

Reply:
Was thinking slice would validate. But then it would need to be totalLength :-/

Will add second .ctor